x/meshsecurity/keeper/keeper.go

```diff
@@ -46,6 +49,47 @@ func NewKeeperX(
 	}
 }
 
+// GetAuthority returns the module's authority.
+func (k Keeper) GetAuthority() string {
+	return k.authority
+}
+
 func (k Keeper) HasMaxCapLimit(ctx sdk.Context, actor sdk.AccAddress) bool {
-	panic("not implemented, yet")
+	store := ctx.KVStore(k.storeKey)
+	return store.Has(types.BuildMaxCapLimitKey(actor))
+}
+
+// GetMaxCapLimit the cap limit is set per consumer contract. Different providers can have different limits
+func (k Keeper) GetMaxCapLimit(ctx sdk.Context, actor sdk.AccAddress) sdk.Coin {
+	return sdk.NewCoin(k.staking.BondDenom(ctx), k.mustLoadInt(ctx, k.storeKey, types.BuildMaxCapLimitKey(actor)))
```

Comment:
There is also the option to store data within the ContractInfoExtension or have a new type. Let's refactor as needed. Protobuf generation and other things are in place already. This will not take long when needed